controller/src/types/instance_status.rs

```diff
@@ -1,24 +1,29 @@
-use serde::{Deserialize,Serialize};
+use serde::{Deserialize, Serialize};
 use validator::Validate;
 
-use crate::client::scheduler::{WorkloadStatus, workload_status::Resources};
+use crate::client::scheduler::{
+    workload_status::{Resources, Status},
+    WorkloadStatus,
+};
 
 #[derive(Debug, Validate, Deserialize, Serialize, Clone)]
 pub struct InstanceStatus {
     #[validate(length(min = 1))]
     pub name: String,
-    pub status_code: i32,
+    pub status_code: InstanceStatusCode,
     pub resource_usage: InstanceResources,
-    pub message: String
 }
 
 impl From<&WorkloadStatus> for InstanceStatus {
     fn from(status: &WorkloadStatus) -> Self {
-        InstanceStatus { 
-            name: status.name.clone(),
-            status_code: status.status_code,
-            resource_usage: InstanceResources { cpu: 1, memory: 1, disk: 1 },
-            message: status.message.clone()
+        InstanceStatus {
+            name: status.instance_id.clone(),
```

Comment:
As here. I think we really didn't go through the ownership issue, as we used many clones to avoid resource access conflicts.

Reply:
I've replaced as many "clone()" as I could with dereferencing (at least for Strings). Not sure if there's a difference in terms of performance, but at least it won't look like we don't know anything other than cloning :)